### PR TITLE
Stabilize integration tests

### PR DESCRIPTION
**Pull request by ethersage**
Improves integration tests in a number of ways:

* Removes the MFE env var in favor of injecting the micro-frontend's config declaratively in each micro-frontend's `webpack.config.js`. Environment variables work fine when running two servers in two process, but we need to be able to use two different MFE ID's in the same process during tests. If we read environment variables inside a module that exports an object it will get cached and we can't get different configs for different frontends.
* Adds a `webpackConfigFactory` to accomplish ☝️ 
* Properly waits to run tests until compilation has completed
* Properly stops the DevServer from watching for file changes in teardown, leaving no remaining async tasks, which lets us remove the `--forceExit` arg to jest
* Increases the test timeout to 30s for steps that rely on compilation or browsing
* Waits for MFE #2 to serve a 200 for its `importManifest.js`
* Shuts webpack up, very chatty
* gets them working again, nbd

TODO:
- [x] Snapshot tests are failing
- [ ] MFE #2 sometimes fails to serve `importManifest.js` at all 😞 
- [ ] Refactor `server.start()` promise wall to be a little more readable

### CODE REVIEW

**Comment by ethersage** [comment]
I'd love some time to take a look at the TODO items in the description, but when you're comfortable to merge, so am I.

**Comment by ethersage** [comment]
Confirmed that babel snapshots are failing on master, as well.

**Comment by ScriptedAlchemy** [comment]
There's some merge conflicts, mind resolving these :) 

Lets just on another slack call when you have a free evening. 
Id like to:
1) Discuss what we have done already
2) go over the issues
3) ask for advice on how I can write better issues 
4) ask for advice on the documentation
5) create some new issues and start labeling them for priority

File: manual/Website1/webpack.config.js [hunk 1]
```diff
@@ -1,24 +1,5 @@
...
-    }),
-    new HtmlWebpackPlugin({
-      template: path.resolve(__dirname, `template.html`),
-      inject: true
-    }),
-  ]
-});
+const siteId = 1;
+
+module.exports = configFactory(siteId);
```
Comment (ScriptedAlchemy) [inline manual/Website1/webpack.config.js]: WAY better!

File: manual/webpack/paths.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
 const path = require('path');
 
-module.exports = {
+module.exports = (siteId) => ({
   root: path.resolve(__dirname, '../'),
   outputPath: path.resolve(__dirname, '../', 'build'),
-  entry: process.env.MFE === 1 ? require.resolve('../Website1/src/index.jsx') : require.resolve('../Website2/src/index.jsx'),
+  entry: siteId === 1 ? require.resolve('../Website1/src/index.jsx') : require.resolve('../Website2/src/index.jsx'),
```
Comment (ScriptedAlchemy) [inline manual/webpack/paths.js]: Really like the approach, appreciate the work to clean this!

File: test/integration/test.js [hunk 3]
```diff
@@ -21,10 +46,12 @@ describe('external script', () => {
     });
 
     it('should console.log', async () => {
-        await page.goto('http://localhost:3001');
+        jest.setTimeout(30000);
```
Comment (ScriptedAlchemy) [inline test/integration/test.js]: omggggg, dude this KILLED me trying to figure it out

**Comment by ethersage** [comment]
@ScriptedAlchemy merge conflicts resolved, tested with prod and dev.

**Pull request merged**